app/tests/components_tests/test_validators.py

```diff
@@ -68,40 +68,43 @@ def test_valid_paths(rel_path):
         ),
     ),
 )
-def test_validate_newick_formats(tree, context):
+def test_validate_newick_format(tree, context):
     with context:
         validate_newick_tree_format(tree=tree)
 
 
 @pytest.mark.parametrize(
-    "error, msg, expected_error",
+    "biom_file, context, msg",
     (
-        (MemoryError, "The file is too large", ValidationError),
-        (SoftTimeLimitExceeded, "The file is too large", ValidationError),
-        (TimeLimitExceeded, "The file is too large", ValidationError),
-        (NewickError, "Invalid Newick tree format", ValidationError),
-        (  # No secrets: only ValidationErrors are returned to user
-            Exception,
-            "",
-            Exception,
+        (
+            # Working example
+            RESOURCE_DIR / "biom" / "valid.biom",
+            nullcontext(),
+            None,
+        ),
+        (
+            # Uncompressed
+            RESOURCE_DIR / "biom" / "uncompressed_OTU_json.biom",
+            pytest.raises(ValidationError),
+            "Only BIOM in valid HDF5 binary file format are supported",
+        ),
+        (
+            # Compressed, but removed first few bytes
+            RESOURCE_DIR / "biom" / "broken.biom",
+            pytest.raises(ValidationError),
+            "Only BIOM in valid HDF5 binary file format are supported",
+        ),
+        (
+            # HD5 Format, but not a biom
+            RESOURCE_DIR / "biom" / "not_a_biom.h5",
+            pytest.raises(ValidationError),
+            "Does not appear to be a BIOM-format file",
         ),
     ),
 )
-def test_validate_newick_exception_handling(
-    error, msg, expected_error, mocker
-):
-
-    class MockParser:
-        @staticmethod
-        def parse(*_, **__):
-            raise error
-
-    mocker.patch(
-        "grandchallenge.components.validators._newick_parser",
-        return_value=MockParser,
-    )
-
-    with pytest.raises(expected_error) as err:
-        validate_newick_tree_format("();")
+def test_validate_biom_format(biom_file, context, msg):
+    with context as c:
+        validate_biom_format(file=biom_file)
 
-    assert msg in str(err)
+    if msg:
```

Comment:
It is preferable to not have logic in tests. Here, it would mean splitting out a test for the happy path and having a separate test for the error cases. I think this is clearer than passing around nullcontext.

Reply:
Agreed, and given that the actual tests are really short and simple (jeej) splitting makes sense.